irc/database.go

```diff
@@ -1008,6 +1008,57 @@ func schemaChangeV19To20(config *Config, tx *buntdb.Tx) error {
 	return nil
 }
 
+// #734: move the email address into the settings object,
+// giving people a way to change it
+func schemaChangeV20To21(config *Config, tx *buntdb.Tx) error {
+	type accountSettingsv21 struct {
+		AutoreplayLines  *int
+		NickEnforcement  NickEnforcementMethod
+		AllowBouncer     MulticlientAllowedSetting
+		ReplayJoins      ReplayJoinsSetting
+		AlwaysOn         PersistentStatus
+		AutoreplayMissed bool
+		DMHistory        HistoryStatus
+		AutoAway         PersistentStatus
+		Email            string
+	}
+	var accounts []string
+	var emails []string
+	callbackPrefix := "account.callback "
+	tx.AscendGreaterOrEqual("", callbackPrefix, func(key, value string) bool {
+		if !strings.HasPrefix(key, callbackPrefix) {
+			return false
+		}
+		account := strings.TrimPrefix(key, callbackPrefix)
+		if _, err := tx.Get("account.verified " + account); err != nil {
+			return true
```

Comment:
Doesn't this key need to be converted for unverified accounts as well?

Reply:
We're relying on buntdb expiration for unverified accounts, so we would want to re-set the value using its original expiration, but buntdb doesn't seem to expose that piece of metadata. In theory I could get this out of the `account.registered.time` key. But I'd rather do it incorrectly.

Reply:
buntdb has an API to get the remaining TTL on a key: https://github.com/tidwall/buntdb/blob/v1.2.6/buntdb.go#L1579

If they key is ignored, users who register on 2.7 (or earlier) then verify on 2.8 will have no email address on their account.